### PR TITLE
Reorganize project to have a single Go module

### PR DESCRIPTION
**Pull request by claudio4j**
Fix https://github.com/apache/camel-k/issues/4686 and #4683

**Release Note**
```release-note
Reorganize project to have a single Go module
```

### CODE REVIEW

File: script/gen_crd/gen-kamelets-crd-api-config.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
             "docsURLTemplate": "https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#{{lower .TypeIdentifier}}-{{arrIndex .PackageSegments -1}}-{{arrIndex .PackageSegments -2}}"
         },
         {
-          "typeMatchPrefix": "^github\\.com/apache/camel-k/pkg/apis/camel/v1\\.",
-          "docsURLTemplate": "xref:apis/camel-k.adoc#_camel_apache_org_v1_{{.TypeIdentifier}}"
+          "typeMatchPrefix": "^github\\.com/apache/camel-k/v2\\.",
```
Comment (squakez) [inline script/gen_crd/gen-kamelets-crd-api-config.json:16]: I don't think this has to be changed, as it related the Kubernetes CRD which is still V1

**Review by squakez — COMMENTED** [review]
Check also the usage of `make set-module-version` which is performed during the release processes.

File: script/gen_crd/gen-kamelets-crd-api-config.json (hunk 1 above)
Comment (claudio4j) [reply script/gen_crd/gen-kamelets-crd-api-config.json]: The `v2` should be prefixed before the `pkg`, so it was fixed to contain the `apis/camel/v1`.
Thanks for the review.

**Comment by oscerd** [comment]
@squakez should be out this week. @christophd can you take a look?

**Comment by claudio4j** [comment]
Once we merge it, there should be a Camel K release, to publish the Camel K 2.x Go modules to go.dev repo.

**Comment by oscerd** [comment]
I can release if needed

**Comment by claudio4j** [comment]
I will backport to the 2.0 release branch and a 2.0.1 release can be done.

**Pull request merged**